### ISSUE TITLE
[bug] Impossible to create a rounded transparent window without borders

### ISSUE THREAD

**Issue by elvinaspredkelis** [tauri-apps/tauri#3481]
### Describe the bug

Before the v1 RC, it was possible to make rounded window borders on macOS by making the window transparent in the `tauri.conf.json`, disabling the decorations and then rounding the borders of the first non-transparent element.

Now, it seems to be impossible to achieve such behavior.

### Reproduction

1. Clone repo: https://github.com/elvinaspredkelis/tauri_1_app
2. Run `tauri dev`

The window has a white background even if the root elements are set to be transparent. And the first element has a rounded border which should make the window appear rounded too.

### Expected behavior

Transparent window bounding to the first non-transparent element in the view.

E.g. Making the `<body>` tag transparent and the first element inside it has `border-radius`. As it's impossible to apply border radius to the `<body>` element. Therefore, it should make the window have rounded borders. 

It was possible to do in earlier releases.

### Platform and versions

```shell
Operating System - Mac OS, version 12.0.1 X64

Node.js environment
  Node.js - 16.6.2
  @tauri-apps/cli - 1.0.0-rc.4
  @tauri-apps/api - 1.0.0-rc.1

Global packages
  npm - 8.3.0
  pnpm - Not installed
  yarn - 1.22.11

Rust environment
  rustup - 1.24.3
  rustc - 1.58.1
  cargo - 1.58.0
  toolchain - stable-x86_64-apple-darwin 

App directory structure
/dist
/node_modules
/public
/src-tauri
/.git
/.vscode
/src

App
  tauri - 1.0.0-rc.2
  tauri-build - 1.0.0-rc.1
  tao - 0.6.2
  wry - 0.13.1
  build-type - bundle
  CSP - default-src 'self'
  distDir - ../dist
  devPath - http://localhost:8080/
  framework - Vue.js
```

**Comment by kebtech** [comment]
Also having this issue in all new RC's. Previously worked fine. I'm also on a Mac and using Vue.js for frontend. I wonder if it is related to the much-loathed vue-cli-plugin-tauri. I noticed that when I delete the entire HTML from WebKit dev tools, there is still a white background. Makes me think the transparent setting in tauri.conf.json is not being honored but not sure how to check that.

**Comment by lucasfernog** [comment]
To use transparency on macOS you need to set https://tauri.studio/docs/api/config#tauri.macOSPrivateApi to true. I'll improve the code to show a warning if this is not set to true.

**Comment by tance77** [comment]
> To use transparency on macOS you need to set https://tauri.studio/docs/api/config#tauri.macOSPrivateApi to true. I'll improve the code to show a warning if this is not set to true.

What is this link?

**Comment by FabianLars** [comment]
yeah not so funny story. we lost that domain so stay clear of tauri.studio links. we currently only serve on tauri.app

**Comment by Zbrooklyn** [comment]
Since this is closed but still ranks for the question, recording what actually works on Windows 11 for anyone landing here.

**For an opaque frameless window, you don't need transparency at all.** Call `DwmSetWindowAttribute` with `DWMWA_WINDOW_CORNER_PREFERENCE` (33) = `DWMWCP_ROUND` (2):

```rust
use windows_sys::Win32::Graphics::Dwm::{DwmSetWindowAttribute, DWMWA_WINDOW_CORNER_PREFERENCE};

let preference: u32 = 2; // DWMWCP_ROUND (1 = DONOTROUND, 3 = ROUNDSMALL)
unsafe {
    DwmSetWindowAttribute(
        hwnd, DWMWA_WINDOW_CORNER_PREFERENCE as u32,
        &preference as *const u32 as *const _,
        std::mem::size_of::<u32>() as u32,
    );
}
```

DWM does the rounding in the compositor, so the corners are genuinely cut — the desktop shows through, antialiased — and you keep the drop shadow. Two caveats from Microsoft's own docs that are worth knowing before you rely on it: the attribute is *"a hint to the system and does not guarantee rounding"*, and windows using per-pixel alpha or a window region **"cannot ever be rounded"**.

**Two things that trip people up on the transparent route this issue is actually about:**

CSS `border-radius` does not round an *opaque* window. The page is painted inside a rectangle, so the radius just exposes whatever is behind it *in the page*. It only rounds the window on `"transparent": true` — and the gotcha that costs the most time is that `html` must be transparent as well as `body`, not just `body`.

And the thing nobody expects: **a custom radius (say 14px) isn't available through a supported API in Electron either.** Electron's `roundedCorners` is a *boolean*, Windows support only landed in v34.3.0, and the request to customise the radius ([electron#47833](https://github.com/electron/electron/issues/47833)) was closed as not planned. Underneath it's the same DWM attribute. So "just use Electron" doesn't buy you a custom radius — the constraint is Windows, not the framework.

Verification note, because this one wasted a day for me: `PrintWindow` renders a window's *own* pixels only and is blind to DWM compositing, so it will happily show you square corners on a window that is actually rounded. You need Windows.Graphics.Capture (what Snipping Tool and OBS use), which reads the visual out of DWM and preserves the alpha channel. An antialiased alpha ramp from ~9 to 255 across six pixels at (0,0) is a rounded corner; a square window reads 255 flat.

Working code, both a frameless template and the capture tool: https://github.com/Zbrooklyn/tauri-snap-layouts
